qiskit_nature/second_q/problems/electronic_properties_container.py

```diff
@@ -0,0 +1,105 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The electronic properties container."""
+
+from __future__ import annotations
+
+from typing import cast
+
+from qiskit_nature.second_q.properties import (
+    AngularMomentum,
+    ElectronicDipoleMoment,
+    Magnetization,
+    ParticleNumber,
+)
+
+from .properties_container import PropertiesContainer
+
+
+class ElectronicPropertiesContainer(PropertiesContainer):
+    """The container class for electronic structure properties."""
+
+    @property
+    def angular_momentum(self) -> AngularMomentum | None:
+        """Returns the :class:`qiskit_nature.second_q.properties.AngularMomentum` property."""
+        return cast(AngularMomentum, self._properties.get("AngularMomentum", None))
+
+    @angular_momentum.setter
+    def angular_momentum(self, angular_momentum: AngularMomentum | None) -> None:
+        """Sets the :class:`qiskit_nature.second_q.properties.AngularMomentum` property."""
+        if angular_momentum is None:
+            self._properties.pop("AngularMomentum", None)
```

Comment:
Wondering about the strings here. Could we use the class name itself, that would save adding extra strings as such. And seeing how all the methods here follow a similar pattern if we had an internal method to which we passed the class it could serve as the implementation for all of these, no?

Reply:
I ran into mypy complaints when trying to do this before. It may just need a proper overloaded variant of the `pop` method though. I will check and get back to you :+1:

Reply:
I extracted an internal `_setter` method to handle this implementation in a central location :+1: